Core/src/Geometry/KDTreeTrackingGeometryBuilder.cpp

```diff
@@ -0,0 +1,217 @@
+          std::make_shared<const RadialBounds>(*radialBounds);
+      auto discLayer =
+          DiscLayer::create(transform, radialBoundsClone, nullptr, 1.);
+      discLayer->assignSurfaceMaterial(surface->surfaceMaterialSharedPtr());
+      tLayer = discLayer;
+    }
+
+  } else if (layerSurfaces.size() > 1u) {
+    // Make a const collection out of the surfaces
+    std::vector<std::shared_ptr<const Surface>> cLayerSurfaces;
+    cLayerSurfaces.reserve(layerSurfaces.size());
+    for (const auto& s : layerSurfaces) {
+      cLayerSurfaces.push_back(s.second);
+    }
+
+    Acts::BinningType bType0 = Acts::equidistant;
+    Acts::BinningType bType1 = Acts::equidistant;
+    std::size_t bins0 = 0;
+    std::size_t bins1 = 0;
+    // In case explicit binning is given
+    if (plVolume.layerSurfaceBinning.size() == 2u) {
+      bType0 = plVolume.layerSurfaceBinning[0u].type;
+      bType1 = plVolume.layerSurfaceBinning[1u].type;
+      // In case explicit bin numbers are given in addition
+      if (bType0 == Acts::equidistant and bType1 == Acts::equidistant and
+          plVolume.layerSurfaceBinning[0u].bins() > 1u and
+          plVolume.layerSurfaceBinning[1u].bins() > 1u) {
+        bins0 = plVolume.layerSurfaceBinning[0u].bins();
+        bins1 = plVolume.layerSurfaceBinning[1u].bins();
+        ACTS_VERBOSE(indent + ">> binning provided externally to be "
+                     << bins0 << " x " << bins1 << ".");
+      }
+    }
+
+    Acts::ProtoLayer pLayer(gctx, cLayerSurfaces);
+    pLayer.envelope = plVolume.extent.envelope();
+    if (plVolume.layerType == Acts::Surface::SurfaceType::Cylinder) {
+      ACTS_VERBOSE(indent + ">> creating cylinder layer with "
+                   << cLayerSurfaces.size() << " surfaces.");
+      // Forced equidistant or auto-binned
+      tLayer = (bins0 * bins1 > 0)
+                   ? m_cfg.layerCreator->cylinderLayer(gctx, cLayerSurfaces,
+                                                       bins0, bins1, pLayer)
+                   : m_cfg.layerCreator->cylinderLayer(gctx, cLayerSurfaces,
+                                                       bType0, bType1, pLayer);
+
+    } else if (plVolume.layerType == Acts::Surface::SurfaceType::Disc) {
+      ACTS_VERBOSE(indent + ">> creating disc layer with "
+                   << cLayerSurfaces.size() << " surfaces.");
+      // Forced equidistant or auto-binned
+      tLayer = (bins0 * bins1 > 0)
+                   ? m_cfg.layerCreator->discLayer(gctx, cLayerSurfaces, bins0,
+                                                   bins1, pLayer)
+
+                   : m_cfg.layerCreator->discLayer(gctx, cLayerSurfaces, bType0,
+                                                   bType1, pLayer);
+    }
+  }
+  if (tLayer != nullptr and tLayer->representingVolume() != nullptr) {
+    ACTS_DEBUG(indent << "> translated into layer bounds: "
+                      << tLayer->representingVolume()->volumeBounds());
+  }
+
+  return tLayer;
+}
```

Comment:
nitpick: new line at the end

Reply:
Inserted, probably we can make `clang-format` to check and fix this?